Show No results message on Quick search and Search Results page

nikhil-narayana-klevu opened the pull request:
Use settings to show the no results found message

rallu reviewed (changes requested):
Missing implementation on keyword specific messages. I added a custom message when keyword is "foobar". 

Also message doesn't show up if first search something normal that gets results and then something that doesn't have any results.

nikhil-narayana-klevu commented:
> Missing implementation on keyword specific messages. I added a custom message when keyword is "foobar".
> 
> Also message doesn't show up if first search something normal that gets results and then something that doesn't have any results.

I added code to reset products when no result found
It works for specific keyword  messages, attaching screenshot. I updated the KMC settings to this
`"noResultsOptions":{
      "showPopularProducts":false,
      "showPopularKeywords":false,
      "messages":[
         {
            "showForTerms":[
               "nikhil"
            ],
            "message":"Nikhil Results"
         },
         {
            "showForTerms":null,
            "message":"No Results!"
         }
      ]
   }
`
![Screenshot from 2023-08-29 16-51-10](https://github.com/klevultd/frontend-sdk/assets/97588976/f1992d22-697d-42da-ad05-863e1e0d03f9)

nikhil-narayana-klevu commented:
@rallu I have updated the code to handle the remaining flags from the KMC settings, you can review it again.

rallu commented on packages/klevu-ui/src/components/klevu-quicksearch/klevu-quicksearch.css:
```diff
@@ -88,3 +89,13 @@ klevu-moi {
+
+.noResultsMessage {
+  text-align: center;
+  border-bottom: 1px solid var(--klevu-color-border, #d6d6d6);
```
There is no such variable as --klevu-color-border. I noticed that there are references to this variable in the codebase. It's legacy. We should use --klevu-color-neutral-3 or --klevu-color-neutral-4 in this case.

rallu replied:
I created separate task to remove --klevu-color-border from the codebase.

rallu commented on packages/klevu-ui/src/global/global.css:
```diff
@@ -21,6 +21,8 @@
   --klevu-color-positive: #2e9c40;
   --klevu-color-negative: #ad2d2d;
 
+  --klevu-color-border: #e0e0e0;
```
remove this

rallu reviewed (approved):
Small changes and its okay.
